zapisy/apps/offer/assignments/views.py

```diff
@@ -65,20 +66,37 @@ def plan_view(request):
         assignments_course_info: AssignmentsCourseInfo = courses[semester][name]
         if group_type not in assignments_course_info:
             assignments_course_info[group_type] = CourseGroupTypeSummary(
-                hours=assignment.hours_semester, teachers=set())
+                hours=assignment.hours_semester, teachers=dict())
         if assignment.teacher_username not in teachers:
             messages.warning(
                 request, f"Użytkownik <strong>{assignment.teacher_username}</strong> "
                 "nie występuje w arkuszu <em>Pracownicy</em>. Przydział został pominięty.")
             continue
         teacher = teachers[assignment.teacher_username]
-        assignments_course_info[group_type].teachers.add(
-            TeacherInfo(username=assignment.teacher_username,
-                        name=f"{teacher.first_name} {teacher.last_name}"))
+        teacher_info = TeacherInfo(username=assignment.teacher_username,
+                                   name=f"{teacher.first_name} {teacher.last_name}")
+        # We do not use a defaultdict since it conflicts with django templates
+        count = assignments_course_info[group_type].teachers.get(teacher_info, 0)
+        assignments_course_info[group_type].teachers[teacher_info] = count + 1
+        hours_to_pensum = assignment.equivalent * assignment.hours_semester / assignment.multiple_teachers
+        stats[semester][group_type] += hours_to_pensum
+        hours_global[semester] += hours_to_pensum
+        # Here we prepare a display string for the hours that count towards pensum
```

Comment:
```suggestion
        
        # We prepare a display string for the hours that count towards pensum.
```

Ten kawałek w ogóle wygląda, jakby chciał się przenieść do jakiejś małej, pomocniczej funkcji. Może do konstruktora tego typu, którego nie mamy?

Wtedy konstruktor mógłby również wykonywać ten rachunek z linijek 81-83.